x-pack/elastic-agent/pkg/agent/application/enroll_cmd.go

```diff
@@ -149,7 +168,95 @@ func NewEnrollCmdWithStore(
 }
 
 // Execute tries to enroll the agent into Fleet.
-func (c *EnrollCmd) Execute() error {
+func (c *EnrollCmd) Execute(ctx context.Context) error {
+	if c.options.FleetServerConnStr != "" {
+		err := c.fleetServerBootstrap(ctx)
+		if err != nil {
+			return err
+		}
+
+		// enroll should use localhost as fleet-server is now running
+		// it must also restart
+		c.options.URL = "http://localhost:8000"
```

Comment:
Yes as it will communicate with the Fleet Server locally. There is currently not a way to setup SSL or run it on a different port through the enroll command.

Definitely things we need to look into, but not in this PR. This is just enough to get it up and running.